src/index.js

```diff
@@ -10,17 +11,20 @@ var expr = module.exports = {
     },
   compiler: function(args, opt) {
       args = args.slice();
+      args.unshift('fn');
       var generator = codegen(opt),
           len = args.length,
           compile = function(str) {
             var value = generator(expr.parse(str));
             args[len] = '"use strict"; return (' + value.code + ');';
-            value.fn = Function.apply(null, args);
+            var generatedFn = Function.apply(null, args);
+            value.fn = generatedFn.bind(null, generator.functionDefinitions);
```

Comment:
http://jsperf.com/function-bind-performance/14

It seems that bind is surprisingly slow. I'll replace it with a closure instead.